### PR TITLE
Stripping and support for forward declarations

### LINKED ISSUE #1: Simplify Carl version?
The Carl version currently has the form `14.22.1-gf9b8b379`. The suffix with the commit hash is not a valid version as far as I can tell. At least the Python version check in a test of pycarl is now failing.
I would propose to restrict the versioning to `14.22.1` which also corresponds to the Storm version scheme.

### PR DESCRIPTION
**Pull request by sjunges**
fixes #1

### CODE REVIEW

File: src/carl/core/MultivariatePolynomial.h [hunk 1]
```diff
@@ -22,6 +23,8 @@
 #include "polynomialfunctions/CoprimePart.h"
 #include "polynomialfunctions/SoSDecomposition.h"
 
+#include "../util/BitVector.h"
```
Comment (volkm) [inline src/carl/core/MultivariatePolynomial.h:26]: use absolute path?

File: src/carl/core/MultivariatePolynomial.tpp [hunk 2]
```diff
@@ -775,7 +775,7 @@ MultivariatePolynomial<C,O,P> MultivariatePolynomial<C,O,P>::remainder(const Mul
 		if(p.lterm().tdeg() < divisor.lterm().tdeg())
 		{
 			assert(!p.lterm().divisible(divisor.lterm()));
-			if( O::degreeOrder )
+			if( OrderedBy::degreeOrder )
```
Comment (volkm) [inline src/carl/core/MultivariatePolynomial.tpp:778]: Need to update template parameter from `O` to `OrderedBy` as well?

**Review by volkm — COMMENTED** [review]
Other than some small remarks, it looks good to me.

**Comment by sjunges** [comment]
Carl does use relative paths throughout its code base.
 It is arguably better to change that (but I would want to do that consistently) but that is not part of this PR :-)

File: src/carl/core/MultivariatePolynomial.tpp (hunk 2 above)
Comment (sjunges) [reply src/carl/core/MultivariatePolynomial.tpp:778]: no, the template parameter is no longer used and instead defaults to an empty struct. I would love to remove the template arguments O and P, but that is one step too much right now

Comment (volkm) [reply src/carl/core/MultivariatePolynomial.tpp:778]: You are right, my mistake.

**Comment by volkm** [comment]
I agree that changing to absolute paths should be part of a different PR.

**Comment by volkm** [comment]
I think we can merge it.

**Comment by sjunges** [comment]
Thanks for your review. 
Let's merge https://github.com/moves-rwth/storm/pull/343 before continueing with this merge. 

I also have to check pycarl. I can look into this tomorrow.

**Pull request merged**